internal/common/conversion/type_conversion.go

```diff
@@ -67,3 +75,14 @@ func MongoDBRegionToAWSRegion(region string) string {
 func AWSRegionToMongoDBRegion(region string) string {
 	return strings.ReplaceAll(strings.ToUpper(region), "-", "_")
 }
+
+type TFPrimitiveType interface {
+	IsUnknown() bool
+}
+
+func NilForUnknown[T any](primitiveAttr TFPrimitiveType, value *T) *T {
```

Comment:
are we doing something similar in other TPF resources? curious why this resource is different

Reply:
[if !plan.Authentication.IsNull() {](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/af46e93db037f1858c3779895aee01a70d115dee/internal/service/streamconnection/model_stream_connection.go#L22)
I think this is usually what is done, splitting the initialization of attributes to multiple lines and using `ifs`